Master - Bug 12065 - queue and state not mandatory

s7design opened the pull request:
http://bugs.otrs.org/show_bug.cgi?id=12065

Hi @mgruner 

Hereby is solved the bug and also we extended the test AgentTicketFreeText.t
We are planing to continue the same with other screens where there are not mandatory fields (Queue, Owner,Responsible,State)
I am not sure if it is necessary for other screens. 

Please let me what do you think about that.

Regards
Zoran

s7design commented:
Hi @mrcbnsls 

You can see we changed in AgentTicketActionCommon.tt ServerError messages. All these messages were 'This field is required.'. Now they are changes similar to ServerError which there is for Service input field. I know that it is not maybe the best solution but it out proposal in this moment. 

There are added mandatory sysconfig items for all screens from AgentTicketActionCommon.pm and AgentTicketMove.pm as well. In my opinion there is no sense for other screens. If you would like we will be able to add for other once.

Selenium test is extended only for AgentTicketFreeText. There is checked mandatory fields. I spoke with Marin by Skype, and we agree that there is not needed extended test with such test cases for other changed screens.

Regards
Zoran

mrcbnsls commented on Kernel/Output/HTML/Templates/Standard/AgentTicketActionCommon.tt:
```diff
@@ -115,6 +114,23 @@ $('#TypeID').bind('change', function (Event) {
+[% END %]
+[% RenderBlockEnd("QueueMandatory") %]
+
 [% RenderBlockStart("Service") %]
```
Hi @s7design, I don't think its necessary to have two different blocks which basically do the same, except for the label marker in case the field is mandatory and some error message. Why not have only one block and decide if it should be mandatory using a parameter which is called on calling Block() in AgentTicketActionCommon.pm?

s7design replied:
Thanks we will change that. You are definitely right.

Zoran

s7design commented:
Hi @mrcbnsls 

I updated modules and removed *Mandatory blocks. I provided solution with TemplateToolKit IF statements. I suppose you like it.
Maybe we should check other screens where there are such mandatory block and I will be able to update it as well in this PR or another one.

Please let me know what do you think about that.

Regards
Zoran

s7design commented:
Hi @mrcbnsls 

It is rebased, I hope it works now

Regards
Zoran

mrcbnsls commented:
For me the state fields now shows up twice in AgentTicketFreeText (and probably in other screens, too).

<img width="310" alt="bildschirmfoto 2016-08-05 um 14 52 57" src="https://cloud.githubusercontent.com/assets/4690172/17437168/5ffdbc18-5b1c-11e6-972f-1ee44bc8397c.png">

s7design commented:
Hi @mrcbnsls 

Thanks, sorry it is my mistake, I've not seen that. It is fixed now, and I hope it is finally right.

Regards
Zoran

mrcbnsls commented:
Thanks, looks good for me. Merged in e93a39c7f38612ba008a4888f5c61449ae1121e7.
